infer/resource.go

```diff
@@ -280,6 +280,10 @@ type Annotator interface {
 	//	mypkg:mymodule:MyResource
 	//
 	SetToken(module, name string)
+
+	AddAlias(module, name string)
+
+	SetResourceDeprecationMessage(message string)
```

Comment:
Nitpicking, but given that this is backed in the implementation by a field called `DeprecationMessage`, I'm inclined to think `Resource` is superfluous here? Moreover, if we're feeling cheeky, we could call it `Deprecate` to align with `Describe`.

Reply:
Yeah, good thoughts .. perhaps also `AddAlias` could just be `Alias`?

Reply:
I called it `SetResourceDeprecationMessage ` because resource properties can also be deprecated. The field `DeprecationMessage` is on the resource level so there's no ambiguity.

For the aliases, I went with "add" because it can be called more than once to add different aliases, which I wanted to call attention to.

Let me know if my reasoning doesn't make sense to you.

Reply:
`Describe` also applies to both the resource and it's fields. I agree that `Deprecate` would have been a more consistent name.